ddtrace/internal/tracemethods.py

```diff
@@ -78,12 +78,24 @@ def _(m):
 
 
 def trace_wrapper(wrapped, instance, args, kwargs):
+    import inspect
+
     from ddtrace.trace import tracer
 
     resource = wrapped.__name__
     if hasattr(instance, "__class__") and instance.__class__ is not type(None):  # noqa: E721
         resource = "%s.%s" % (instance.__class__.__name__, resource)
 
+    # Check for async
+    if inspect.iscoroutinefunction(wrapped):
+
+        async def async_wrapper(*a, **kw):
```

Comment:
Open question: Since this will have a noticeable impact on span duration for existing async applications using this configuration, do you think it's worth wrapping this around a config so people have to opt into the new treatment (with the new span duration)?

Reply:
I guess if we imagine the potential support tickets that would arise from making this change without a config option, that would help us get a sense of whether a config would be necessary. From my understanding it seems like this is a clear improvement of the alignment of behavior with expecation, so we'd have a convincing story to tell in response to those questions.

Reply:
I think the durations change is a good thing. The current behavior is broken and doesn't provide a lot of value. I am not a fan of shipping bug fixes behind a feature flag.

Reply:
Thanks for the discussion points. If we don't hide it behind a config flag, we should be very clear that this will have an impact of span duration and any metrics that may be computed from these spans.